src/client/app/components/AdminComponent.jsx

```diff
@@ -3,15 +3,129 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import React from 'react';
+import axios from 'axios';
+import { FormControl, Button } from 'react-bootstrap';
+import { chartTypes } from '../reducers/graph';
+import HeaderContainer from '../containers/HeaderContainer';
 
-/**
- * React component that controls the Admin panel
- * @returns JSX to create the AdminComponent
- */
-export default function AdminComponent() {
-	return (
-		<div>
-			<p>Admin panel</p>
-		</div>
-	);
+export default class AdminComponent extends React.Component {
+	constructor(props) {
+		super(props);
+		this.handleDisplayTitleChange = this.handleDisplayTitleChange.bind(this);
+		this.handleDefaultChartToRenderChange = this.handleDefaultChartToRenderChange.bind(this);
+		this.handleDefaultBarStackingChange = this.handleDefaultBarStackingChange.bind(this);
+		this.handleSubmitPreferences = this.handleSubmitPreferences.bind(this);
+	}
+
+	handleDisplayTitleChange(e) {
+		this.props.updateDisplayTitle(e.target.value);
+	}
+
+	handleDefaultChartToRenderChange(e) {
+		this.props.updateDefaultGraphType(e.target.value);
+	}
+
+	handleDefaultBarStackingChange() {
+		this.props.toggleDefaultBarStacking();
+	}
+
+	handleSubmitPreferences() {
+		axios.post('/api/preferences',
+			{
+				token: localStorage.getItem('token'),
+				preferences: {
+					displayTitle: this.props.displayTitle,
+					defaultChartToRender: this.props.defaultChartToRender,
+					defaultBarStacking: this.props.defaultBarStacking
+				}
+			})
+			.then(() => {
+				this.props.showNotification({
+					message: 'Updated preferences',
+					level: 'success',
+					position: 'tr',
+					autoDismiss: 3
+				});
+			})
+			.catch(() => {
+				this.props.showNotification({
+					message: 'Failed to submit changes',
+					level: 'error',
+					position: 'tr',
+					autoDismiss: 3
+				});
```

Comment:
You have this 'show notification' logic everywhere, and the only parameters that differ are the level and the message.
These should be factored out into functions that take in a message.